server/store/store_test.go

```diff
@@ -1311,24 +1312,28 @@ func testLockAndUnlockWhitelist(t *testing.T, store *SQLStore, _ *plugintest.API
 
 	currentTime := time.Now()
 	ch := make(chan int, 1)
+	wg := sync.WaitGroup{}
+	wg.Add(1)
 	go func() {
+		defer wg.Done()
 		tx, txErr := store.BeginTx()
 		assert.Nil(txErr)
 
-		err = store.LockWhitelist(tx)
-		assert.Nil(err)
+		txErr = store.LockWhitelist(tx)
+		assert.Nil(txErr)
 
 		ch <- 1
 		time.Sleep(1 * time.Second)
-		err = store.UnlockWhitelist(tx)
-		assert.Nil(err)
+		txErr = store.UnlockWhitelist(tx)
+		assert.Nil(txErr)
 
-		err = store.CommitTx(tx)
-		assert.Nil(err)
+		txErr = store.CommitTx(tx)
+		assert.Nil(txErr)
 	}()
 
 	<-ch
 	_, err = store.GetSizeOfWhitelist(nil)
 	assert.Nil(err)
 	assert.Greater(time.Since(currentTime), 1*time.Second)
+	wg.Wait()
```

Comment:
why are we not applying this before fetching the size of the whitelist?

Reply:
We don't want the whole goroutine to finish when we are fetching the size of the whitelist. We are putting this at the end so that the other goroutine can finish before out test ends.